helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -82,15 +83,15 @@
     _remainingCapacity = new HashMap<>(instanceCapacity);
     _maxPartition = clusterConfig.getMaxPartitionsPerInstance();
     _currentAssignedReplicaMap = new HashMap<>();
-    _highestCapacityUtilization = 0;
+    _highestCapacityUtilization = 0f;
   }
 
   /**
    * This function should only be used to assign a set of new partitions that are not allocated on
    * this node.
    * Using this function avoids the overhead of updating capacity repeatedly.
    */
-  void assignNewBatch(Collection<AssignableReplica> replicas) {
+  void assign(Collection<AssignableReplica> replicas) {
```

Comment:
This method can only be used as initialization. Please don't change the name so users won't be confused.

Reply:
I'm not making it more complicated but making it more consistent without breaking anything right? Do you think renaming a method would change anything?

Reply:
Please read my comment. The current logic CANNOT be used as a regular batch assignment. There is a potential bug.

Reply:
Got it after sync up offline. I've added a TODO here as it's not obviously seen at first look.